### DISCUSSION TITLE
Waiting to run: ...

### DISCUSSION THREAD

**Discussion by Freed-Wu** [zellij-org/zellij#3607] [Q&A] [answered]
When I try to quit zellij, then start it again, I will meet

```
Waiting to run: sh /home/wzy/.local/share/zinit/plugins/zsh-nix-shell/scripts/buildShellShim --rcfile /tmp/nix-shell-8826-0/rc
```

What is its meaning? Why I meet it?

TIA!

**Comment by Lockszmith-GH** [comment]
This looks like some issue with zsh's zinit plugins, not Zellij. Have you reached out to their community about this?

**Reply by Freed-Wu** [reply]
It not only one script, there are other similar cases.

**Reply by Lockszmith-GH** [reply]
"waiting to run" is a zinit message while it is initializing.
It will probably show up in every pane you open, because zsh is your default shell.

That's why the key to debugging it is focusing on zsh.

You'll need to debug your .zshrc script - maybe some environment variable aren't properly setup

Without Zellij, after you've logged on and in the shell, what happens when you run `zsh -l` (which opens a new login shell)?

**Reply by Freed-Wu** [reply]
I run `exec zellij` in zshrc. Then for example, I edit a file with vi (nvim), then I quit and press `<C-D>` to quit zsh, zellij will quit, too. Then I open terminal emulator, I will get:

```
Waiting to run: /run/current-system/sw/bin/vi -u /nix/store/l6i8282myrnad5lf2h1phlrmc7ax8dkq-init.lua --cmd lua vim.g.node_host_prog='/nix/store/zkfcgjyq2mvl3qpi3jb46krwp9xb0jip-neovim-0.10.1/bin/nvim-node';vim.g.loaded_perl_provider=0;vim.g.loaded_python_provider=0;vim.g.python3_host_prog='/nix/store/zkfcgjyq2mvl3qpi3jb46krwp9xb0jip-neovim-0.10.1/bin/nvim-python3';vim.g.ruby_host_pr<ENTER> run, <ESC> drop to shell, <Ctrl-c> exit-neovim-0.10.1/bin/nvim-ruby' tools/nixd.cpp
```

**Reply by Lockszmith-GH** [reply, marked as answer]
ok, I see now what you're describing.
Sorry for being so dense.

That's the session resurrection feature.

see: https://zellij.dev/documentation/session-resurrection
#2919 is probably what you're looking for.
